docs/openshift/deploy.md

```diff
@@ -0,0 +1,113 @@
+
+## ACS Engine the Long Way
+
+### Step 1: Generate an SSH Key
+
+In addition to using OpenShift APIs to interact with the clusters, cluster operators may access the master machine using SSH.
+
+If you don't have an SSH key [cluster operators may generate a new one](../ssh.md#ssh-key-generation).
+
+### Step 2: Create a Service Principal
+
+acs-engine needs a Service Principal to interact with Azure Resource Manager (ARM). Follow the instructions to [create a new service principal](../serviceprincipal.md).
+
+### Step 3: Edit your Cluster Definition
+
+ACS Engine consumes a cluster definition which outlines the desired shape, size, and configuration of OpenShift. There are a number of features that can be enabled through the cluster definition: check the `examples` directory for a number of... examples.
+
+Edit the [simple OpenShift cluster definition](/examples/openshift.json) and fill out the required values (every value with empty default `""` must be filled in):
+
+* `openShiftConfig`: contains authentication info for OpenShift. You can provide either `clusterUsername` and `clusterPassword` for simple htpasswd authentication, or set the `enableAADAuthentication` to `true` to enable AAD integration between OpenShift and Azure. AAD Authentication is preferred and more secure option, but requires more prerequisites and privileges
+* `azProfile`: Azure account information. `subscriptionId` and `tenantId` can be obtained from `$ az account list -o json` command. `resourceGroup` and `location` should be filled in based on your preference
+* `masterProfile`:
+  * `count`: currently only one master is supported
+  * `dnsPrefix`: must be a region-unique name and will form part of the hostname (e.g. myprod1, staging, leapingllama)
+  * `vmSize`: must be at least `Standard_D4s_v3`
+* `agentPoolProfiles`: contains specification for infrastructure and compute nodes
+  * `vmSize`: must be at least `Standard_D4s_v3`
+* `linuxProfile`: `keyData` must contain the public portion of an SSH key - this will be associated with the `adminUsername` system user (e.g. 'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABA....')
+* `clientId`: this is the service principal's appId uuid or name from step 2
+* `secret`: this is the service principal's password or randomly-generated password from step 2
+
+### Step 4: Generate the Templates
+
+The generate command takes a cluster definition and outputs a number of templates which describe your OpenShift cluster. By default, `generate` will create a new directory named after your cluster nested in the `_output` directory. If my dnsPrefix was `openshift-red` my cluster templates would be found in `_output/openshift-red-`.
```

Comment:
s/a number of templates which describe/a template and parameters file which describes/